lib/scraper.js

```diff
@@ -26,24 +26,49 @@ function loadHtmlAndCss (context, po) {
 
 function Scraper (options) {
 	this.originalResources = [];
+	this.occupiedFileNames = [];
 	this.loadedResources = [];
+	this.respondedResourcePromises = {};
+	this.loadedResourcePromises = {};
 
 	this.options = _.extend({}, defaults, options);
 	this.options.request = _.extend({}, defaults.request, options.request);
 	this.options.directory = path.resolve(process.cwd(), this.options.directory || '');
 	this.options.filenameGenerator = getFilenameGenerator(this.options.filenameGenerator);
 }
 
-Scraper.prototype.getLoadedResource = function getLoadedResource (resource) {
-	return _.find(this.loadedResources, function checkUrlsEqual (lr) {
-		return compareUrls(resource.getUrl(), lr.getUrl());
-	});
+Scraper.prototype.addOccupiedFileName = function addOccupiedFileName (filename) {
+	this.occupiedFileNames.push(filename);
+};
+
+Scraper.prototype.getOccupiedFileNames = function getOccupiedFileNames () {
+	return this.occupiedFileNames;
 };
 
 Scraper.prototype.addLoadedResource = function addLoadedResource (resource) {
 	this.loadedResources.push(resource);
 };
 
+Scraper.prototype.addRespondedResourcePromise = function addRequestedResourcePromise (url, promise) {
```

Comment:
Scraper.prototype.add**Responded**ResourcePromise, but function add**Requested**ResourcePromise

so requested or responded? (the same in the next method)